bin/hdfcoinc/pycbc_coinc_hdfinjfind

```diff
@@ -74,113 +76,197 @@ if args.verbose:
 
 fo = h5py.File(args.output_file, 'w')
 injection_index = 0
-for trigger_file, injection_file in zip(args.trigger_files, args.injection_files):   
+for trigger_file, injection_file in zip(args.trigger_files,
+                                        args.injection_files):
     logging.info('Read in the coinc data: %s' % trigger_file)
     f = h5py.File(trigger_file, 'r')
 
+    #detect whether trigger file is two-ifo or multi-ifo style
+    if 'foreground/time1' in f:
+        multi_ifo_style = False
+    else:
+        multi_ifo_style = True
+        # get list of groups which contain subgroup 'time' - these will be the IFOs
+        ifo_list = [key for key in f['foreground'] if 'time' in f['foreground/%s/' % key]]
+
     template_id = f['foreground/template_id'][:]
     stat = f['foreground/stat'][:]
     ifar_exc = f['foreground/ifar_exc'][:]
     ifar = f['foreground/ifar'][:]
     fap = f['foreground/fap'][:]
     fap_exc = f['foreground/fap_exc'][:]
-    time1 = f['foreground/time1'][:]
-    time2 = f['foreground/time2'][:]
-    trig1 = f['foreground/trigger_id1'][:]
-    trig2 = f['foreground/trigger_id2'][:]
-    ana_start = f['segments/coinc/start'][:]
-    ana_end = f['segments/coinc/end'][:]
-    time = 0.5 * (time1 + time2)
+    if not multi_ifo_style:
+        # Using the original, non-multiifo-style organisation
+        time1 = f['foreground/time1'][:]
+        time2 = f['foreground/time2'][:]
+        trig1 = f['foreground/trigger_id1'][:]
+        trig2 = f['foreground/trigger_id2'][:]
+        ana_start = f['segments/coinc/start'][:]
+        ana_end = f['segments/coinc/end'][:]
+        time = 0.5 * (time1 + time2)
+    else:
+        ifo_times = ()
+        time_dict = {}
+        trig_dict = {}
+        for ifo in ifo_list:
+             ifo_times += (f['foreground/%s/time' % ifo][:],)
+             time_dict[ifo] = f['foreground/%s/time' % ifo][:]
+             trig_dict[ifo] = f['foreground/%s/trigger_id' % ifo][:]
+        time = numpy.array([pycbc.events.mean_if_greater_than_zero(vals)[0]
+                            for vals in zip(*ifo_times)])
+        # The injections missed because of 'off' time will be in
+        # 'no-coincident-detector' time, so need to combine segments
+        # to give the time when any two detectors are on
+        any_seg = segments.segmentlist([])
+        for key in f:
+            if key == 'foreground':
```

Comment:
can't you just iterate over keys in f['segments'] ?

Reply:
I forgot that this bit was there - it shouldn't actually have the segments key if it is using the new style file - as it will not have the segments group. The group will now be the different detector combos, and so I can remove this elif statement safely